masonry/src/lib.rs

```diff
@@ -100,6 +100,8 @@
 #![cfg_attr(not(debug_assertions), allow(unused))]
 // False-positive with dev-dependencies only used in examples
 #![cfg_attr(not(test), warn(unused_crate_dependencies))]
+// Needed because of the SelfMut macro
+#![allow(clippy::needless_arbitrary_self_type)]
 
```

Comment:
It's not clear how this can work without a `#![feature(arbitrary_self_types)]` (probably in a `cfg_attr`)

Reply:
To be clear, this lint doesn't need `#![feature(arbitrary_self_types)]`. It applies to cases like:

```rust
fn foobar(self: Self) {}
```

which have a more concise version.

Reply:
I wasn't commented on the added line - I was commenting on "the top of `masonry/src/lib.rs`", because it isn't clear to me how the new docs can work without enabling the feature for `std::ops::Receiver`.
Like, they do seem to work based on local testing, but I'm really struggling to work out how.

I think it's a rustdoc bug.